src/common/markdown.ts

```diff
@@ -159,3 +159,40 @@ export const checkHasMention = (content: string, username: string) => {
     return words.some((word) => word === `@${username}`);
   });
 };
+
+export const findMarkdownTag = ({
+  tokens,
+  tag,
+  depth = 0,
+  maxDepth = 2,
+}: {
+  tokens: Token[];
+  tag: string;
+  depth?: number;
+  maxDepth?: number;
+}): Token | undefined => {
+  if (depth > maxDepth) {
+    return undefined;
+  }
```

Comment:
since this is recursive (due to markdown structure being tree) I added depth so we don't go too deep and affect performance.

By default its 2 level, which should be more then enough for markdown content we can create in our editor.